docs/object-capabilities/derived-authority.md

```diff
@@ -0,0 +1,147 @@
+# Delegating and restricting authority
+
+> ## [Plan]
+>
+> * demonstrate basic use in combination with file system access etc.
+> * show how a derived authority can be constructed
+> * refer to the patterns for more complex scenarios
+>
+> ## [TODO]
+>
+> * [ ] come up with a better title for the section
+> * [ ] toc stuff and linking
+
+Any interesting program will need to interact with the outside world, like accessing the network or the file system, or by creating and communicating with other programs.
+We call __ambient authority__ all those rights implicitly granted to the program to make these things possible, and because Pony makes this concept *explicit*, we need to take
+the time to talk about what it means and how it works. In other languages like for example C, you can always attempt to do an operation like I/O, and it will usually succeed
+save some runtime checks (your disk being full might make an operation fail, for example). But in Pony, any piece of code interacting with the outside world needs the authority to do so.
+
+The operating system essentially knows nothing about the structure of any program it runs, in particular it is ignorant of any Pony specific concepts.
+In order to impose that some code is authorized for an operation like writing to disk, Pony requires a special argument to be passed, a capability bearing the authority required for the
+task at hand. The type system guarantees that no code that has not been passed this capability can perform the task requiring authorization, so this kind of safety is guaranteed at compile time.
```

Comment:
```suggestion
task at hand. The type system guarantees that inadequate capabilities for a task fail at compile time.
```

Purpose here being to remove the double-negative of "**no** code that has **not** been passed".

Reply:
In addition it talks about passing the wrong capabiity, where I only thought about "none". Thanks!